src/types/artifact.ts

```diff
@@ -0,0 +1,33 @@
+import { ArtifactType, ResultArtifact, ResultArtifactData, MarkdownArtifact, MarkdownArtifactData, UnknownArtifact, Artifact, ArtifactData, TableArtifact, TableArtifactData } from '@/models/Artifact'
+
+export function isResultArtifact(artifact: Artifact): artifact is ResultArtifact {
+  return artifact.type === ArtifactType.Result
+}
+
+export function isMarkdownArtifact(artifact: Artifact): artifact is MarkdownArtifact {
+  return artifact.type === ArtifactType.Markdown
+}
+
+export function isTableArtifact(artifact: Artifact): artifact is TableArtifact {
+  return artifact.type === ArtifactType.Table
+}
+
+export function isKnownArtifactType(type: string): type is ArtifactType {
+  return Object.values(ArtifactType).includes(type as ArtifactType)
+}
+
+export function isUnknownArtifact(artifact: Artifact): artifact is UnknownArtifact {
+  return artifact.type === ArtifactType.Unknown
+}
```

Comment:
These are a little odd to me. ArtifactType includes `unknown` as a type. So `isKnownArtifactType` will return true if the type being checked is `unknown`. And `isUnknownArtifact` will return false if a valid type is passed. Both of which make sense on their own. But in context of each other they appear to be inverse checks but they're not.

Reply:
Ah that helps. I think then all I would do is change the `isKnownArtifactType` to `type !== ArtifactType.unknown` Since the mapper is intentionally narrowing the type from `string` to `ArtifactType`.